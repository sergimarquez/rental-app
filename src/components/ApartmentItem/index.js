import React, { Component } from 'react';
import { format } from "date-fns";
import { Row, Col,  Card } from "react-bootstrap";

export default class AparmentItem extends Component {
  render() {
    const { apartment } = this.props;
    var date = new Date(apartment.date);
    var formatDate = format(date, "MMMM d, yyyy");

    return (
      <Card className="mb-4">
        <Card.Header>
          <h2 className="h4">{apartment.name}</h2>
        </Card.Header>
        <Card.Body>
          <Row>
            <Col sm>
              <p>{apartment.description}</p>
              <p>
                <strong>Price: </strong>
                {apartment.price}$
              </p>
              <p>
                {apartment.isAvailable ? (
                  <span className="text-success">Available</span>
                ) : (
                  <span className="text-danger">Rented</span>
                )}
              </p>
            </Col>

            <Col sm>
              <p>
                <strong>Floor size: </strong>
                {apartment.floor}mÂ²
              </p>
              <p>
                <strong>Rooms: </strong> {apartment.rooms}
              </p>
              <p>
                <strong>Date added: </strong> {formatDate}
              </p>
              <p>
                <strong>Realtor: </strong>
                {apartment.realtor}
              </p>
            </Col>
          </Row>
        </Card.Body>
      </Card>
    );
  }
}