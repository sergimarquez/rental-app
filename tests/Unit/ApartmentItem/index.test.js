import React from 'react';
import renderer from 'react-test-renderer';
import ApartmentItem from "../../../src/components/ApartmentItem";

test('apartment gets rendered', () => {
  const testApartment = {
    name: 'Test Flat',
    description: 'Bright flat in the center',
    price: 140,
    isAvailable: true,
    rooms: 4,
    realtor: 'John Doe',
    date: new Date(),
  }
  const component = renderer.create(<ApartmentItem apartment={testApartment} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('apartment with different values', () => {
  const testApartment = {
    name: 'House with a view',
    description: 'Panoramic views from this mansion',
    price: 800,
    isAvailable: false,
    rooms: 5,
    realtor: 'Jane Doe',
    date: new Date(),
  }
  const component = renderer.create(<ApartmentItem apartment={testApartment} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
