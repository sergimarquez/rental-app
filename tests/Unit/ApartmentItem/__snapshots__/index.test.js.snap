// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apartment gets rendered 1`] = `
<div
  className="mb-4 card"
>
  <div
    className="card-header"
  >
    <h2
      className="h4"
    >
      Test Flat
    </h2>
  </div>
  <div
    className="card-body"
  >
    <div
      className="row"
    >
      <div
        className="col-sm"
      >
        <p>
          Bright flat in the center
        </p>
        <p>
          <strong>
            Price: 
          </strong>
          140
          $
        </p>
        <p>
          <span
            className="text-success"
          >
            Available
          </span>
        </p>
      </div>
      <div
        className="col-sm"
      >
        <p>
          <strong>
            Floor size: 
          </strong>
          m²
        </p>
        <p>
          <strong>
            Rooms: 
          </strong>
           
          4
        </p>
        <p>
          <strong>
            Date added: 
          </strong>
           
          April 20, 2020
        </p>
        <p>
          <strong>
            Realtor: 
          </strong>
          John Doe
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`apartment with different values 1`] = `
<div
  className="mb-4 card"
>
  <div
    className="card-header"
  >
    <h2
      className="h4"
    >
      House with a view
    </h2>
  </div>
  <div
    className="card-body"
  >
    <div
      className="row"
    >
      <div
        className="col-sm"
      >
        <p>
          Panoramic views from this mansion
        </p>
        <p>
          <strong>
            Price: 
          </strong>
          800
          $
        </p>
        <p>
          <span
            className="text-danger"
          >
            Rented
          </span>
        </p>
      </div>
      <div
        className="col-sm"
      >
        <p>
          <strong>
            Floor size: 
          </strong>
          m²
        </p>
        <p>
          <strong>
            Rooms: 
          </strong>
           
          5
        </p>
        <p>
          <strong>
            Date added: 
          </strong>
           
          April 20, 2020
        </p>
        <p>
          <strong>
            Realtor: 
          </strong>
          Jane Doe
        </p>
      </div>
    </div>
  </div>
</div>
`;
